### PR TITLE
Add a configureInitialCongestionWindow method to Envoy::Network::Connection

### PR DESCRIPTION
**Pull request by wu-bin**
Add a configureInitialCongestionWindow method to Envoy::Network::Connection to set the initial congestion window.

QUIC connections will use QuicConnection::AdjustNetworkParameters to set
the congestion window.
TCP connections will delegate to the TransportSocket, which is a no-op
by default because there is no portable api to set TCP congestion
window. An application can use a custom TransportSocket and override the
method to set the congestion window in a proprietary way.

Signed-off-by: Bin Wu <wub@google.com>

Commit Message: Add a configureInitialCongestionWindow method to Envoy::Network::Connection to set the initial congestion window.
Risk Level: Low.
Testing: None, just adding an interface that is not used in OSS envoy. I'll test it in my application tests.

### CODE REVIEW

**Review by danzh2010 — COMMENTED** [review]
LGTM overall. @mattklein123 for interface changes.

File: envoy/network/connection.h [hunk 1]
```diff
@@ -326,6 +326,17 @@ class Connection : public Event::DeferredDeletable,
...
+
+  /**
+   * Try to configure the connection's initial congestion window.
+   * The operation is advisory - the connection may not support it, even if it's supported, it may
+   * not do anything after the first few network round trips with the peer.
+   * @param bandwidth_bits_per_sec The estimated bandwidth between the two endpoints of the
+   * connection.
+   * @param rtt The estimated round trip time between the two endpoints of the connection.
+   */
+  virtual void configureInitialCongestionWindow(uint64_t bandwidth_bits_per_sec,
```
[the remark this answers is not in the document]
Comment (wu-bin) [reply envoy/network/connection.h:342]: Sure, done.

**Review by mattklein123 — DISMISSED** [review]
LGTM, thanks.

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**